File: src/main/kotlin/dev/jefy/connectpro/portfolio/applicaion/query/ProjectQuery.java
package dev.jefy.connectpro.portfolio.applicaion.query;

import org.jspecify.annotations.NullMarked;

import java.util.List;

import dev.jefy.connectpro.portfolio.applicaion.dtos.ProjectResponse;
import dev.jefy.connectpro.portfolio.domain.vo.PortfolioId;

/**
 * @author Jôph Yamba
 */
@NullMarked
public interface ProjectQuery {
    ProjectResponse get(ProjectId projectId);
    List<ProjectResponse> getAllForPortfolio(PortfolioId portfolioId);
}

File: src/main/kotlin/dev/jefy/connectpro/portfolio/application/query/ProjectQuery.kt
package dev.jefy.connectpro.portfolio.application.query

import dev.jefy.connectpro.portfolio.application.dtos.ProjectResponse
import dev.jefy.connectpro.portfolio.domain.vo.PortfolioId
import dev.jefy.connectpro.portfolio.domain.vo.ProjectId

interface ProjectQuery {
    fun get(projectId: ProjectId): ProjectResponse
    fun getAllForPortfolio(portfolioId: PortfolioId): List<ProjectResponse>
}